fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -617,10 +616,6 @@ where
 
     pub(crate) fn run_program(&mut self) -> Result<ProgramState, InterpreterError> {
         loop {
-            if self.registers[RegId::PC] >= VM_MAX_RAM {
```

Comment:
How do we know that it is not possible? Were you able to check all the places modifying the PC register?

Also maybe worth to put it under `debug_assert` condition for future changes

Reply:
Even better, I reworked the instruction fetching so that `Interpreter::execute` checks the memory boundary when doing an instruction fetch. This means we can close https://github.com/FuelLabs/fuel-vm/issues/38 as well.